Mathlib/LinearAlgebra/AffineSpace/Independent.lean

```diff
@@ -941,6 +942,61 @@ theorem reindex_map {m n : ℕ} (s : Simplex k P m) (e : Fin (m + 1) ≃ Fin (n
     (s.map f hf).reindex e = (s.reindex e).map f hf :=
   rfl
 
+section restrict
+attribute [local instance] AffineSubspace.toAddTorsor
+
+/-- Restrict an affine simplex to an affine subspace that contains it. -/
+@[simps]
+def restrict {n : ℕ} (s : Affine.Simplex k P n) (S : AffineSubspace k P)
+    (hS : affineSpan k (Set.range s.points) ≤ S) :
```

Comment:
Would this be better or worse as
```suggestion
    (hS : ∀ i, s.points i ∈ S) :
```
The current spelling is nice because you can just pass `le_rfl` for what I suspect is a common use-case.

Reply:
I also suspect `le_rfl` is the most common case (and the most common other case probably still has equality, just not definitional equality - if you e.g. have multiple simplices spanning the same subspace).

Reply:
> (i.e., similar to `LinearMap.codLift` rather than `LinearMap.codRestrict`, which I claim is inferior). What do you think?

I claim the opposite, since `codRestrict` is computable!

Reply:
As usual, the computability argument cuts very little ice with me but I won't insist on my suggestion since this is clearly subjective.